driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -140,19 +142,31 @@ public BsonDocument getResumeToken() {
     /**
      * Returns the namespace
      *
+     * The invalidate operation type does include a MongoNamespace in the ChangeStreamDocument response. The
+     * dropDatabase operation type includes a MongoNamespace, but does not include a collection name as part
+     * of the namespace.
+     *
      * @return the namespace
+     * @throws IllegalStateException if the namespaceDocument does not contain the 'db' or 'coll' keys
```

Comment:
Looking at this again, this looks wrong.  If the property is Nullable, and we're going to return null when there is no namespaceDocument, shouldn't we just do the same if either db or coll is missing?

If you agree, remove the `@throws` annotation and add an `@return` indicating when null will be returned. And change the tests to match.

Reply:
I think that's would be most consistent. I also changed `getDatabaseName()` to follow this same pattern.